
syntax = "proto3"; 

package traack.v1.auth;

import "proto/v1/response/status.proto";
import "google/protobuf/timestamp.proto";

option go_package = "github.com/mhannig/traack/service/lib/v1/auth";

message AuthToken {
    string bearer = 10;
    google.protobuf.Timestamp issued_at = 20;    
    google.protobuf.Timestamp expires_at = 30;
}


message AuthenticateApiKeyRequest {
    string api_key = 10;
    string api_secret = 20;
}


message AuthenticateApiKeyResponse {
    AuthToken auth_token = 10;
    traack.v1.response.Status status = 20;
}

service TokenService {
    rpc AuthenticateApiKey(AuthenticateApiKeyRequest)
        returns (AuthenticateApiKeyResponse) {};
}

